flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/connector/elasticsearch/sink/BulkRequestConsumerFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.elasticsearch.sink;
+
+import org.apache.flink.annotation.Internal;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.client.RestHighLevelClient;
+
+import java.io.Serializable;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+/**
+ * {@link BulkRequestConsumerFactory} is used to bridge incompatible Elasticsearch Java API calls
+ * across different Elasticsearch versions.
+ */
+@Internal
+interface BulkRequestConsumerFactory
+        extends Function<
+                        RestHighLevelClient, BiConsumer<BulkRequest, ActionListener<BulkResponse>>>,
```

Comment:
Do you gain anything by extending `Function`? I'd rather go with an explicit `create`.
An alternative cut may be to actually add an explicit `BulkRequestConsumer` (or handler?) class and use `SerializableFunction<RestHighLevelClient,  BulkRequestConsumer>` as the factory.

Reply:
I'm not quite sure I understand, SerializableFunction also extends Function

Reply:
In the end I opted for the interface with `create`